src/validate.js

```diff
@@ -2429,30 +2429,31 @@ function F_VALIDATE(disabledHL) {
 
 			// check Public connection
 			if (slowChecks
-				&& segment.$isRoutable){
+				&& segment.$isRoutable
+				&& nodeA.$otherSegments.length > 0
+				&& nodeB.$otherSegments.length > 0){
 				// Check other segments to be a drivable public segment
 				var foundPublicConnection = false;
-				for (var i = 0; i < nodeA.$otherSegmentsLen; i++) {
+				for (var i = 0; i < nodeA.$otherSegments.length; i++) {
 					var otherSegment = nodeA.$otherSegments[i];
-					if (otherSegment.$isRoutable){
+					if (otherSegment.$rawSegment.isRoutable()){
 						foundPublicConnection = true;
 						break;
 					}
 				}
-				// if not already found, check node B
-				if(!foundPublicConnection){
```

Comment:
This `if()` looks fine to me, i.e. we should avoid nodeB checks if there a public connection found on nodeA. Or did I miss something?